scripts/common.sh

```diff
@@ -313,13 +313,21 @@ pre_minion_checks() {
 get_salt_container() {
     local -r max_retries=10
     local salt_container='' attempts=0
+    local -a found_containers=()
 
-    while [ -z "$salt_container" ] && [ $attempts -lt $max_retries ]; do
-        salt_container="$(crictl ps -q \
+    while [[ $attempts -lt $max_retries ]]; do
+        IFS=$'\n' read -r -d '' -a found_containers < <(crictl ps -q \
             --label io.kubernetes.pod.namespace=kube-system \
             --label io.kubernetes.container.name=salt-master \
-            --state Running)"
+            --state Running && printf '\0')
```

Comment:
Why not simply do:
```
 readarray -t found_containers < <(crictl ps -q \
            --label io.kubernetes.pod.namespace=kube-system \
            --label io.kubernetes.container.name=salt-master \
            --state Running)
```

Reply:
I've read somewhere that `readarray` and `mapfile` could have both been used, but both depend on Bash version, while my approach should be POSIX compliant. True? If not, I can change to `readarray`.

Reply:
It's true, `readarray` comes in bash 4, but since we're using bashism everywhere, I don't think it's really important to be POSIX compliant.
And we're deploying MetalK8s only on systems that ship bash version 4+.
But anyway, we can keep your approach, my point was more I don't understand why your chaning `IFS` and use `printf`, it should be sufficient to do something like:
```
        read -r -d '' -a found_containers < <(crictl ps -q \
            --label io.kubernetes.pod.namespace=kube-system \
            --label io.kubernetes.container.name=salt-master \
            --state Running)
```